tokio-console/src/view/controls.rs

```diff
@@ -0,0 +1,141 @@
+use crate::view::{self, bold};
+
+use once_cell::sync::OnceCell;
+use tui::{
+    layout,
+    text::{Span, Spans, Text},
+    widgets::{Paragraph, Widget},
+};
+
+/// Construct a widget to display the controls available to the user in the
+/// current view.
+pub(crate) struct Controls {
+    paragraph: Paragraph<'static>,
+    height: u16,
+}
+
+impl Controls {
+    pub(in crate::view) fn new(
+        view_controls: &Vec<ControlDisplay>,
+        area: &layout::Rect,
+        styles: &view::Styles,
+    ) -> Self {
+        let universal_controls = universal_controls();
+
+        let mut spans_controls = Vec::with_capacity(view_controls.len() + universal_controls.len());
+        spans_controls.extend(view_controls.iter().map(|c| c.to_spans(styles)));
+        spans_controls.extend(universal_controls.iter().map(|c| c.to_spans(styles)));
+
+        let mut lines = vec![Spans::from(vec![Span::from("controls: ")])];
+        let mut current_line = lines.last_mut().expect("This vector is never empty");
+        let separator = Span::from(", ");
+
+        let controls_count: usize = spans_controls.len();
+        for (idx, spans) in spans_controls.into_iter().enumerate() {
+            if idx == 0 || current_line.width() == 0 {
+                current_line.0.extend(spans.0);
+            } else {
```

Comment:
style nit, take it or leave it: i might prefer
```suggestion
                current_line.0.extend(spans.0);
                continue;
            }
```
here, so that the majority of the code in this loop, which is currently an `else` block, doesn't need to be indented? but, this is up to you.

Reply:
Done (I didn't apply the suggestion because I had to re-indent everything anyway).